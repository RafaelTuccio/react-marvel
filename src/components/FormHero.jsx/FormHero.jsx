import { useState } from 'react'
import ic_busca from '../assets/ic_busca.svg'

const FormHero = (props) => {
    let [name, setName] = useState("")
    let {heroHandle} = props
    return(
        <form className="form-hero"
        onSubmit={(e) => {
            e.preventDefault();
        }}
        >
            <input type="text" 
                placeholder="Procure por um herÃ³i"
                value={name}
                onChange={(e) =>{
                    setName(e.target.value)
                    heroHandle("hulk")
                }}
            />
            <img src={ic_busca} alt="icone de lupa" />
        </form>
    );
}

export default FormHero;