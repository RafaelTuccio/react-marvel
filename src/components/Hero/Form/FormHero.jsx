import { useState } from 'react'
import { useHero } from '../../../service/useHero'

import ic_busca from '../../assets/ic_busca.svg'

const FormHero = (props) => {
    let [name, setName] = useState("")
    let [error, setError] = useState("")
    let {heroHandle, setAutorizacao, setLoad} = useHero();
    return(
        <div>
            <form className="form-hero"
                onSubmit={(e) => {
                e.preventDefault();
                if(name){
                    heroHandle({name});
                    setLoad(true)
                    setName("")
                    setAutorizacao(false)
                }else{
                    setError("digite o nome de um herói para começar a procurar!!!")
                }
                
            }}>
                <input type="text" 
                    placeholder="Procure por um herói"
                    value={name}
                    onChange={(e) => {
                        setName(e.target.value);
                        setError(null)
                    }}
                />
                <img src={ic_busca} alt="icone de lupa" />
            </form>
            <div className="box-error">
                <span className="error">{error}</span>
            </div>
        </div>
    );
}

export default FormHero;