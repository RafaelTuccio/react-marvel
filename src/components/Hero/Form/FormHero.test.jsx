import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import FormHero from './FormHero';

describe("<FormHome />", () => {
    it("deve renderizar normalmente na tela", () => {
        const {debug} = render(<FormHero />)
        debug()
    })

    it("deve chamar a função onChange toda ver que uma tecla for pressionada no input", () => {
        const fn = jest.fn();
        render(<FormHero heroHandle={fn}/>)
        const input = screen.getByPlaceholderText(/Procure por um herói/i)
        const value = "spider-man,{enter}";

        userEvent.type(input, value)
        
        expect((fn)).toHaveBeenCalledTimes(1)
    })

    it("deve dar match com o snapshot", () => {
        const fn = jest.fn();
        const { container } = render(<FormHero heroHandle={fn}/>);
        expect(container.firstChild).toMatchSnapshot();
    })



    
})