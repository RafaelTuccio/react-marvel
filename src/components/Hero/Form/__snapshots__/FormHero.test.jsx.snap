// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHome /> deve dar match com o snapshot 1`] = `
<form
  class="form-hero"
>
  <input
    placeholder="Procure por um herÃ³i"
    type="text"
    value=""
  />
  <img
    alt="icone de lupa"
    src="ic_busca.svg"
  />
</form>
`;
